eclair-core/src/main/scala/fr/acinq/eclair/Features.scala

```diff
@@ -137,91 +155,99 @@ object Features {
     }
   }
 
-  case object OptionDataLossProtect extends Feature {
+  case object OptionDataLossProtect extends Feature with InitFeature with NodeFeature {
     val rfcName = "option_data_loss_protect"
     val mandatory = 0
   }
 
-  case object InitialRoutingSync extends Feature {
+  case object InitialRoutingSync extends Feature with InitFeature {
     val rfcName = "initial_routing_sync"
     // reserved but not used as per lightningnetwork/lightning-rfc/pull/178
     val mandatory = 2
   }
 
-  case object OptionUpfrontShutdownScript extends Feature {
+  case object OptionUpfrontShutdownScript extends Feature with InitFeature with NodeFeature {
     val rfcName = "option_upfront_shutdown_script"
     val mandatory = 4
   }
 
-  case object ChannelRangeQueries extends Feature {
+  case object ChannelRangeQueries extends Feature with InitFeature with NodeFeature {
     val rfcName = "gossip_queries"
     val mandatory = 6
   }
 
-  case object VariableLengthOnion extends Feature {
+  case object VariableLengthOnion extends Feature with InitFeature with NodeFeature with InvoiceFeature {
     val rfcName = "var_onion_optin"
     val mandatory = 8
   }
 
-  case object ChannelRangeQueriesExtended extends Feature {
+  case object ChannelRangeQueriesExtended extends Feature with InitFeature with NodeFeature {
     val rfcName = "gossip_queries_ex"
     val mandatory = 10
   }
 
-  case object StaticRemoteKey extends Feature {
+  case object StaticRemoteKey extends Feature with InitFeature with NodeFeature {
     val rfcName = "option_static_remotekey"
     val mandatory = 12
   }
 
-  case object PaymentSecret extends Feature {
+  case object PaymentSecret extends Feature with InitFeature with NodeFeature with InvoiceFeature {
     val rfcName = "payment_secret"
     val mandatory = 14
   }
 
-  case object BasicMultiPartPayment extends Feature {
+  case object BasicMultiPartPayment extends Feature with InitFeature with NodeFeature with InvoiceFeature {
     val rfcName = "basic_mpp"
     val mandatory = 16
+    val init = true
+    val nodeAnnouncement = true
+    val invoice = true
```

Comment:
This is no longer needed.

Reply:
Good catch, thanks!

Reply:
Fixed in https://github.com/ACINQ/eclair/pull/2063/commits/bffbec2c81ac25357da22d1556057d5a7e24e3bb